sunpy/coordinates/wcs_utils.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+import astropy.wcs.utils
+
+from .frames import *
+
+__all__ = ['solar_wcs_frame_mapping']
+
+def solar_wcs_frame_mapping(wcs):
+    """
+    This function registers the coordinates frames to their FITS-WCS coordinate
+    type values in the `astropy.wcs.utils.wcs_to_celestial_frame` registry.
+    """
+
+    xcoord = wcs.wcs.ctype[0][0:4]
+    ycoord = wcs.wcs.ctype[1][0:4]
```

Comment:
Do you want to support cases where x and y are swapped? In Astropy, we're forcing the axes to be re-ordered for this function with:

```
    wcs = wcs.sub([WCSSUB_CELESTIAL])
```

but since I'm not sure if you can do this here, you may want to hard code trying both orders?

What if you have a spectral cube? For Astropy, this still works, again because of the above.